### PR TITLE
Rpath addition.

### PR DESCRIPTION
**Pull request by woodruffw**
Broken off from #40. ~~WIP until #38 is merged.~~

cc @UniqMartin

### CODE REVIEW

**Comment by woodruffw** [comment]
I went ahead and implemented `MachOFile#add_command` in this PR, since `add_rpath`'s behavior is a subset of it.

File: test/test_tools.rb [hunk 1]
```diff
@@ -146,11 +146,52 @@ def test_change_rpath_fat
...
+      MachO::Tools.add_rpath(actual, "/foo/bar/baz")
+
+      original = MachO::MachOFile.new(filename)
+      modified = MachO::MachOFile.new(actual)
+
+      assert_operator modified.ncmds, :>, original.ncmds
+      assert_operator modified.sizeofcmds, :>, original.sizeofcmds
+      assert_operator modified.rpaths.size, :>, original.rpaths.size
+      refute_includes original.rpaths, "/foo/bar/baz"
+      assert_includes modified.rpaths, "made_up_path"
```
Comment (woodruffw) [inline test/test_tools.rb]: Whoops, this should also be `"/foo/bar/baz"`. I'll fix this when I'm home.

File: test/test_tools.rb [hunk 2]
```diff
@@ -146,11 +146,52 @@ def test_change_rpath_fat
...
+
+    groups.each do |filename, actual|
+      FileUtils.cp filename, actual
+      MachO::Tools.add_rpath(actual, "/foo/bar/baz")
+
+      original = MachO::FatFile.new(filename)
+      modified = MachO::FatFile.new(actual)
+
+      refute_includes original.rpaths, "/foo/bar/baz"
+      assert_includes modified.rpaths, "made_up_path"
```
Comment (woodruffw) [inline test/test_tools.rb]: Same here.

File: test/test_tools.rb (hunk 1 above)
Comment (UniqMartin) [inline test/test_tools.rb]: So what's wrong with the test if this mistake didn't cause the test run to fail?

Comment (woodruffw) [inline test/test_tools.rb]: It's just testing the wrong string - `"made_up_path"` was already in each file, `"/foo/bar/baz"` is the one that was added and should be tested for.

Comment (UniqMartin) [inline test/test_tools.rb]: Oh, right! This is all about addition so of course the pre-existing RPATH is still included.

Comment (woodruffw) [inline test/test_tools.rb]: Yep, just a careless copy-paste :smile:

File: lib/macho/macho_file.rb [hunk 3]
```diff
@@ -170,6 +170,33 @@ def command(name)
...
     alias :[] :command
 
+    # Add a new load command to the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being added
+    # @return [void]
+    # @raise [MachO::LoadCommandNotSerializableError] if the command can't be serialized
+    # @raise [MachO::HeaderPadError] if the new command exceeds the header pad buffer
+    # @note This is public, but methods like {#add_rpath} should be preferred.
+    def add_command(lc)
+      raise LoadCommandNotSerializableError.new(LOAD_COMMANDS[lc]) unless lc.serializable?
```
Comment (UniqMartin) [inline lib/macho/macho_file.rb]: Is this check actually needed? I think it is redundant as the `LoadCommand#serialize` invocation below will perform the exact same check.

**Comment by UniqMartin** [comment]
One minor remark, but otherwise this looks good to me. :+1:

File: lib/macho/macho_file.rb (hunk 3 above)
Comment (woodruffw) [inline lib/macho/macho_file.rb]: Nope, just an oversight. I'll fix it.

**Comment by woodruffw** [comment]
> One minor remark, but otherwise this looks good to me. :+1:

Great! Merging now...

**Comment by UniqMartin** [comment]
Awesome! :tada:

I think the next steps would/should be:

- Re-implement `MachOFile#change_rpath` to use load command serialization.
- Re-implement `MachOFile#change_dylib_id` to use load command serialization.
- Re-implement `MachOFile#change_install_name` to use load command serialization.
- Drop no longer needed `MachOFile#set_lc_str_in_cmd`.

Am I right? Anything else that needs to be addressed beforehand?

**Comment by woodruffw** [comment]
Yep, that sounds right. I'll open a PR for these tonight :smile:

**Pull request merged**